### PR TITLE
Properly slugify the alias on Project Relationships.

### PR DESCRIPTION
**Pull request by ericholscher**
**This includes a migration that will break some URL's**.
In the past we haven't slugified aliases,
so this will break URL's for projects in that state.

This shows effected projects:

```
(<ProjectRelationship: tests-docs -> tests-docs-spanish>, u'/es/latest/', u'eslatest')
(<ProjectRelationship: test_bcn_cn -> test_bcn_cn>, u'/cn', u'cn')
(<ProjectRelationship: recharge_api_calls -> recharge_api_calls>, u'/recharge', u'recharge')
```

### CODE REVIEW

**Comment by ericholscher** [comment]
Pared down the initial migration to not change capitalization, and it looks like it does what I want. People will break their own URL's going forward if they try and change their alias.

File: readthedocs/projects/migrations/0022_add-alias-slug.py [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+        ),
+        migrations.AlterField(
+            model_name='project',
+            name='has_valid_webhook',
+            field=models.BooleanField(default=False, help_text='This project has been built with a webhook'),
+        ),
+        migrations.AlterField(
+            model_name='project',
+            name='programming_language',
+            field=models.CharField(blank=True, choices=[('words', 'Only Words'), ('py', 'Python'), ('js', 'JavaScript'), ('php', 'PHP'), ('ruby', 'Ruby'), ('perl', 'Perl'), ('java', 'Java'), ('go', 'Go'), ('julia', 'Julia'), ('c', 'C'), ('csharp', 'C#'), ('cpp', 'C++'), ('objc', 'Objective-C'), ('other', 'Other')], default='words', help_text='The primary programming language the project is written in.', max_length=20, verbose_name='Programming Language'),
```
Comment (ericholscher) [inline readthedocs/projects/migrations/0022_add-alias-slug.py]: Above here is just noise from things we've missed I think? I can probably kill it.

File: readthedocs/projects/models.py [hunk 2]
```diff
@@ -52,7 +52,7 @@ class ProjectRelationship(models.Model):
                                related_name='subprojects')
     child = models.ForeignKey('Project', verbose_name=_('Child'),
                               related_name='superprojects')
-    alias = models.CharField(_('Alias'), max_length=255, null=True, blank=True)
+    alias = models.SlugField(_('Alias'), max_length=255, null=True, blank=True)
```
Comment (humitos) [inline readthedocs/projects/models.py]: I think `db_index=False` should be added. Since it's `True` by default for `SlugField`

**Comment by ericholscher** [comment]
It runs automatically in Django. Django's allow's `_`, which is fine for this use case.

**Comment by ericholscher** [comment]
screenshot 2017-12-21 15 22 56

**Pull request merged**